src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -0,0 +1,596 @@
+#include "postgres.h"
+
+#include "access/htup_details.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_proc.h"
+#include "commands/defrem.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/version_compat.h"
+#include "lib/stringinfo.h"
+#include "nodes/makefuncs.h"
+#include "nodes/nodes.h"
+#include "nodes/value.h"
+#include "parser/parse_func.h"
+#include "parser/parse_type.h"
+#include "utils/builtins.h"
+#include "utils/guc.h"
+#include "utils/lsyscache.h"
+#include "utils/syscache.h"
+
+/* forward declaration for deparse functions */
+static void AppendAlterFunctionStmt(StringInfo buf, AlterFunctionStmt *stmt);
+static void AppendDropFunctionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendFunctionName(StringInfo buf, ObjectWithArgs *func);
+static void AppendFunctionNameList(StringInfo buf, List *objects);
+
+static void AppendDefElem(StringInfo buf, DefElem *def);
+static void AppendDefElemStrict(StringInfo buf, DefElem *def);
+static void AppendDefElemVolatility(StringInfo buf, DefElem *def);
+static void AppendDefElemLeakproof(StringInfo buf, DefElem *def);
+static void AppendDefElemSecurity(StringInfo buf, DefElem *def);
+static void AppendDefElemParallel(StringInfo buf, DefElem *def);
+static void AppendDefElemCost(StringInfo buf, DefElem *def);
+static void AppendDefElemRows(StringInfo buf, DefElem *def);
+static void AppendDefElemSet(StringInfo buf, DefElem *def);
+
+static void AppendRenameFunctionStmt(StringInfo buf, RenameStmt *stmt);
+static void AppendAlterFunctionSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt);
+static void AppendAlterFunctionOwnerStmt(StringInfo buf, AlterOwnerStmt *stmt);
+static void AppendAlterFunctionDependsStmt(StringInfo buf, AlterObjectDependsStmt *stmt);
+
+
+/*
+ * DeparseAlterFunctionStmt builds and returns a string representing the AlterFunctionStmt
+ */
+const char *
+DeparseAlterFunctionStmt(AlterFunctionStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	AppendAlterFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendAlterFunctionStmt appends a string representing the AlterFunctionStmt to a buffer
+ */
+static void
+AppendAlterFunctionStmt(StringInfo buf, AlterFunctionStmt *stmt)
+{
+	ListCell *actionCell = NULL;
+
+#if (PG_VERSION_NUM < 110000)
+	appendStringInfo(buf, "ALTER FUNCTION ");
+#else
+	if (stmt->objtype == OBJECT_FUNCTION)
+	{
+		appendStringInfo(buf, "ALTER FUNCTION ");
+	}
+	else
+	{
+		appendStringInfo(buf, "ALTER PROCEDURE ");
+	}
+#endif
+
+	AppendFunctionName(buf, stmt->func);
+
+	foreach(actionCell, stmt->actions)
+	{
+		DefElem *def = castNode(DefElem, lfirst(actionCell));
+		AppendDefElem(buf, def);
+	}
+
+	appendStringInfoString(buf, ";");
+}
+
+
+/*
+ * AppendDefElem appends a string representing the DefElem to a buffer
+ */
+static void
+AppendDefElem(StringInfo buf, DefElem *def)
+{
+	if (strcmp(def->defname, "strict") == 0)
```

Comment:
Are we sure that we'd always have lower case-letters in defelem? If not, we could consider using `pg_strncasecmp`. If yes, can you please change some of the tests to have CapiTal lETters in the elems?

Reply:
This is exactly how it is used in Postgres codebase. So I think it is perfetly safe to use them this way.

For reference:
https://github.com/postgres/postgres/blob/f5daf7f3266ff5a92f1bf8d386bd5ac3d7d042d6/src/backend/commands/functioncmds.c#L494

Reply:
I will however add some tests with mixed letters like 
`ALTER FUNCTION "CiTuS.TeeN"."TeeNFuNc.1!?!"() StrIct`
 to be safe

Reply:
side note: I updated the logic to uppercase all the DefElem values in the final deparsed query. So this `strict` keyword will be `STRICT` in the final query string